### PR TITLE
Support preinstall Prometheus server

### PR DESCRIPTION
**Pull request by Jeffwan**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Address https://github.com/kubernetes/perf-tests/issues/1713

`--enable-prometheus-server` will help us provision prometheus stack in the cluster. However,  It's hard for it to bring up prometheus server due to environment variance.  This PR allows CL2 to scratch metrics from self-deployed Prometheus server and it also shift prometheus setup responsibility to user. 

Default behavior remains same and this will be enabled by explicitly setting corresponding arguments.  `--preinstall-prometheus-server`.  Following arguments are used to construct prometheus config so that query executor know which services to talk with. 

```
--preinstall-prometheus-service-namespace
--preinstall-prometheus-service-name
--preinstall-prometheus-service-port
```

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Support preinstall Prometheus server in CL2 tests
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

Usage example:

```
PROMETHEUS_STORAGE_CLASS_PROVISIONER=kubernetes.io/aws-ebs PROMETHEUS_STORAGE_CLASS_VOLUME_TYPE=gp2
ETCD_CERTIFICATE=/etc/kubernetes/pki/kube-apiserver/etcd-client.crt ETCD_KEY=/etc/kubernetes/pki/kube-apiserver/etcd-client.key KUBE_SSH_USER=ubuntu AWS_SSH_KEY=/Users/jiaxin/.ssh/id_rsa 
go run cmd/clusterloader.go --kubeconfig=/Users/jiaxin/.kube/config --testconfig=/Users/jiaxin/go/src/k8s.io/perf-tests/clusterloader2/testing/density/config.yaml --provider=aws  --alsologtostderr --v=2 --report-dir=/tmp/pt/ --nodes=2 
--preinstall-prometheus-server 
--preinstall-prometheus-service-namespace=prometheus 
--preinstall-prometheus-service-name=prometheus --preinstall-prometheus-service-port=9095

```

### CODE REVIEW

**Comment by Jeffwan** [comment]
Let me resolve the failures and update the pr

**Comment by mm4tt** [comment]
/assign @marseel 

Marcel, could you take a look?

File: clusterloader2/cmd/clusterloader.go [hunk 1]
```diff
@@ -290,6 +293,13 @@ func main() {
 		if err := prometheusController.SetUpPrometheusStack(); err != nil {
 			klog.Exitf("Error while setting up prometheus stack: %v", err)
 		}
+	} else if clusterLoaderConfig.PrometheusConfig.PreInstallServer {
+		if prometheusController, err = prometheus.NewController(&clusterLoaderConfig); err != nil {
```
Comment (marseel) [inline clusterloader2/cmd/clusterloader.go:297]: I'm just wondering about one thing. We heavily use prometheus-operator. In SetUpPrometheusStack we apply manifests that create PodMonitors/ServiceMonitors, you can check them here: https://github.com/kubernetes/perf-tests/tree/master/clusterloader2/pkg/prometheus/manifests/default
In your case though, we can't do it so some of the features of CL2 won't be working, unless you configure preinstalled prometheus manually. Is this something that is fine with your use case?

**Comment by marseel** [comment]
/cc @mborsz 
Could you please take a look and approve if everything is fine?

**Comment by marseel** [comment]
/cc @jkaniuk 
Could you take a look and approve if everything is fine?

File: clusterloader2/README.md [hunk 2]
```diff
@@ -162,6 +162,20 @@ Allows to scrape metrics from all pods in service. Here you can find example [Se
...
 
+### Use preinstall Prometheus server
+
+If you already have a prometheus server to use in the cluster, you can ask CL2 to scrape metrics from that target server
+instead of setting up new prometheus and deleting it after the test.
+
+Following options are used to enable preinstall prometheus server and specify service details
+
+```
+--preinstall-prometheus-server
```
Comment (jkaniuk) [inline clusterloader2/README.md:173]: Could we replace all those flags with one that passes info and enables functionality?

--preinstall-prometheus-server-url:<service>.<namespace>.svc.cluster.local:<port>

Comment (Jeffwan) [reply clusterloader2/README.md:173]: Good point, I think that's reasonable. I will make the change

Comment (marseel) [reply clusterloader2/README.md:173]: Did you maybe have time to look into this?

**Pull request closed without merging**